commit ea40bf88bce04fb212cc946d8e006d2731a5d944
Author: Damien Lespiau <damien.lespiau@intel.com>
Date:   Wed Jul 25 14:51:09 2012 +0100

    tests: Use pkg-config to find setbacklight CFLAGS and libs
    
    setbacklight depends on udev and drm, one cannot just add the -l flags
    if you want to support unusual install paths.

diff --git a/configure.ac b/configure.ac
index c35d987..a8fc7f2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -224,6 +224,7 @@ fi
 AC_CHECK_PROG(RSVG_CONVERT, rsvg-convert, rsvg-convert)
 AM_CONDITIONAL(HAVE_RSVG_CONVERT, test -n "$RSVG_CONVERT")
 
+PKG_CHECK_MODULES(SETBACKLIGHT, [libudev libdrm])
 
 if test "x$GCC" = "xyes"; then
 	my_common_gcc_flags="-Wall -Wextra -Wno-unused-parameter \
diff --git a/tests/Makefile.am b/tests/Makefile.am
index b788977..1f5514f 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -35,6 +35,7 @@ setbacklight_SOURCES =				\
 	$(top_srcdir)/src/libbacklight.c	\
 	$(top_srcdir)/src/libbacklight.h
 
-setbacklight_LDADD = -ludev -ldrm
+setbacklight_CFLAGS = $(AM_CFLAGS) $(SETBACKLIGHT_CFLAGS)
+setbacklight_LDADD = $(SETBACKLIGHT_LIBS)
 
 EXTRA_DIST = weston-test
